includes/class-sensei-messages.php

```diff
@@ -523,6 +519,16 @@ public function save_new_message_post( $sender_id = 0, $receiver_id = 0, $messag
 				add_post_meta( $message_id, '_posttype', $post->post_type );
 				add_post_meta( $message_id, '_post', $post->ID );
 
+				/**
+				 * Fires when a new private message is sent.
+				 *
+				 * @since 1.8.0
+				 * @hook  sensei_new_private_message
+				 *
+				 * @param {int} $message_id The message ID.
+				 */
+				do_action( 'sensei_new_private_message', $message_id );
```

Comment:
In ancient times, the contact teacher was a static form and when you submit it it would redirect to the same url with some additional query params in it. Something like  `?message_submitted=true`. Which would show the notice to the user that the message was submitted successfully.

When we needed to add the AJAX version of the form on top of it, we needed to prevent the redirection when the message was submitted, otherwise the ajax response would not work properly. So I moved this hook outside of `save_new_message_post` method so I could use it for the ajax version [here](https://github.com/Automattic/sensei/blob/8fcb0dc96a5cbe9432caeb051f654ba2087b9098/includes/rest-api/class-sensei-rest-api-send-message-controller.php#L120)

But it looks like I was oblivious to the fact that the ajax version of the send message api also needs to fire that hook so emails are sent. Looks like I introduced this bug. Sorry 🙂 

Does it still work, with the redirect? Does the message being sent without page refresh?

Reply:
You are absolutely right - there was an unnecessary redirect request after making the submit request in learning mode. I didn't notice it at first because it's not actually redirecting, I think that's because it's an AJAX request.

I've fixed it by changing the redirect logic a bit. Let me know what you think. 🙂